packages/pwa/app/hooks/use-navigation.js

```diff
@@ -26,7 +27,22 @@ const useNavigation = () => {
          * @param  {...any} args - additional args passed to `.push` or `.replace`
          */
         (path, action = 'push', ...args) => {
-            const updatedHref = buildPathWithUrlConfig(path, {locale})
+            // sanitize the locale/site from the current relative url
+            let [pathname, search] = path.split('?')
+            pathname = pathname.replace(`/${site.alias || site.id}`, '').replace(`/${locale}`, '')
+            search = search
```

Comment:
A couple questions here: 

1. There are alias's for locales, so is the patch or search is using a locale alia's will the replace not find a match?
2. In the place where you are replacing the site id or alias on line 32, is it possible that the path includes a site id, but we are trying to replace an occurrence of the site alias, meaning there will be no replacement? 

I'm wondering is there will be a scenario that leads to the above where say a user bookmarked a page during a time when the configuration didn't have an alias defined for a site or locale. That link shouldn't break anything, and I think it might in this case. I think any links that use id's for site and locale should always work even if the website it configured to use alias's. 

You might be able to fix this by updating how you sanitize, and ensure that both id's and alias's are always removed. 

It would be nice to see some tests for this hook to ensure that things are working as intended.

Reply:
Hey @bendvc, I've removed the sanitization logic out of this hook since it is not necessary since the path we pass into this hook is always a path without any site and locale ref, and that is how it is supposed to use
```
navigate('/login')
```
The reason I added it in the first place was there was only one exception when we want to redirect people from the account page to the login page if they are not signed in. As you have pointed out using the `navigation` hook during rendering is not a good practice, the hook is not being used for that case anymore. Hence we don't really need to sanitize step.